Utils/microsoft_api.py

```diff
@@ -0,0 +1,300 @@
+            'access_token': access_token,
+            'valid_until': time_now + expires_in,
+        }
+
+        if refresh_token:
+            integration_context['current_refresh_token'] = refresh_token
+        demisto.setIntegrationContext(integration_context)
+        return access_token
+
+    def _oproxy_authorize(self):
+        content = self.refresh_token or self.tenant_id
+        oproxy_response = requests.post(
+            self.token_retrieval_url,
+            json={
+                'app_name': self.app_name,
+                'registration_id': self.auth_id,
+                'encrypted_token': self.get_encrypted(content, self.enc_key)
+            },
+            verify=self.use_ssl
+        )
+
+        if oproxy_response.status_code not in {200, 201}:
+            msg = 'Error in authentication. Try checking the credentials you entered.'
+            try:
+                demisto.info('Authentication failure from server: {} {} {}'.format(
+                    oproxy_response.status_code, oproxy_response.reason, oproxy_response.text))
+                err_response = oproxy_response.json()
+                server_msg = err_response.get('message')
+                if not server_msg:
+                    title = err_response.get('title')
+                    detail = err_response.get('detail')
+                    if title:
+                        server_msg = f'{title}. {detail}'
+                if server_msg:
+                    msg += ' Server message: {}'.format(server_msg)
+            except Exception as ex:
+                demisto.error('Failed parsing error response - Exception: {}'.format(ex))
+            raise Exception(msg)
+        try:
+            gcloud_function_exec_id = oproxy_response.headers.get('Function-Execution-Id')
+            demisto.info(f'Google Cloud Function Execution ID: {gcloud_function_exec_id}')
+            parsed_response = oproxy_response.json()
+        except ValueError:
+            raise Exception(
+                'There was a problem in retrieving an updated access token.\n'
+                'The response from the Oproxy server did not contain the expected content.'
+            )
+
+        return (parsed_response.get('access_token'), parsed_response.get('expires_in', 3595),
+                parsed_response.get('refresh_token'))
+
+    def _get_self_deployed_token(self):
+        if not self.app_url:
+            url = f'https://login.windows.net/{self.tenant_id}/oauth2/token'
+        else:
+            url = self.app_url
+        data = {
+            'client_id': self.client_id,
+            'client_secret': self.client_secret,
+            'grant_type': 'client_credentials'
+        }
+
+        if self.scope:
+            data['scope'] = self.scope
+        else:
+            data['resource'] = self.resource
+
+        body: dict = {}
+        try:
+            response = requests.post(url, data, verify=self.use_ssl)
+            if response.status_code != 200:
```

Comment:
lets be consistent on the way we check status code, here we check `!= 200` and somewhere else we check if its in a set